admin/admin.py

```diff
@@ -210,6 +210,9 @@ def on_model_change(self, form, model, is_created):
             # document has just been validated for the first time
             model.validation_time = datetime.datetime.now()
             got_new_file = model.has_file
+        elif model.validation_time is not None and not model.validated:
+            # document is no longer validated => reset validation time
+            model.validation_time = None
```

Comment:
Okay, that should work. Should we remove the `validated` column then? We can turn it into a property equal to `validation_time is not None`.